### PR TITLE
Add Item: Java Set

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Add Java Set entry from Target Keyword list.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/set/set.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Set` is an interface provided in `java.util` that implements collections that have no duplicate elements. Some implementations will have restrictions on the elements the `Set` can contain, such as excluding `null` elements. Adding an ineligible element will typically throw an unchecked `NullPointerException` or a `ClassCastException`.
```
Comment (KTom101) [inline content/java/concepts/set/set.md:14]:
```suggestion
A `Set` is an interface provided in the `java.util` package that implements collections that have no duplicate elements. Some implementations will have restrictions on the elements the `Set` can contain, such as excluding `null` elements. Adding an ineligible element will typically throw an unchecked `NullPointerException` or a `ClassCastException`.
```

File: content/java/concepts/set/set.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Syntax
+
+```pseudo
+SetClass<DataType> mySet = new SetClass<DataType>
+```
+
+The `SetClass` is a class that implements the `Set` interface, and is initialized with a generic type inside angle brackets `< ... >`.
+
```
Comment (KTom101) [inline content/java/concepts/set/set.md:23]: Can there be an example of a set being created?

File: content/java/concepts/set/set.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+## Methods
+
+The `Set` interface utilizes the following methods:
+
+- `.add()` : `.add(item)` adds `item` to the `Set` if it is not already present.
+- `.clear()` : Removes all the elements from the `Set`
+- `.contains()` : `.contains(item)` returns `true` if `item` is a member of the `Set`.
+- `.remove()` : `.remove(item)` will remove `item` from the `Set`.
+- `.size()` : Returns the number of items in the `Set`.
```
Comment (KTom101) [inline content/java/concepts/set/set.md:32]: Can a couple of these methods be shown in an example? Almost like how it was done in the [javascript sets entry ](https://www.codecademy.com/resources/docs/javascript/sets).

File: content/java/concepts/set/set.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Set` is an interface provided in `java.util` that implements collections that have no duplicate elements. Some implementations will have restrictions on the elements the `Set` can contain, such as excluding `null` elements. Adding an ineligible element will typically throw an unchecked `NullPointerException` or a `ClassCastException`.
+
+## Syntax
+
+```pseudo
+SetClass<DataType> mySet = new SetClass<DataType>
```
Comment (KTom101) [inline content/java/concepts/set/set.md:19]:
```suggestion
SetClass<DataType> mySet = new SetClass<DataType> ();
```

File: content/java/concepts/set/set.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+The `SetClass` is a class that implements the `Set` interface, and is initialized with a generic type inside angle brackets `< ... >`.
```
Comment (KTom101) [inline content/java/concepts/set/set.md:22]: Can you briefly mention the different set classes that can be used like HashSet, TreeSet, and LinkedHashSet?

**Review by KTom101 — CHANGES_REQUESTED** [review]
@SSwiniarski I've finished the first review of this.

**Comment by SSwiniarski** [comment]
@KTom101 I added a `HashSet` example to address your comments.

**Review by KTom101 — APPROVED** [review]
@Dusch4593 @yangc95 This is ready for a second review

File: content/java/concepts/set/set.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+   food.add("Salad");
+   food.remove("Sausage");
+   System.out.println(food.contains("Sausage"));
+   System.out.println(food);
+   System.out.println(food.size());
+ }
+}
+```
+
+This will output:
```
Comment (Dusch4593) [inline content/java/concepts/set/set.md:60]:
```suggestion
This will output the following:
```

Comment (Dusch4593) [inline content/java/concepts/set/set.md:60]:
```suggestion
This will output the following:
```

File: content/java/concepts/set/set.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Set` is an interface provided in the `java.util` package that implements collections that have no duplicate elements. Some implementations will have restrictions on the elements the `Set` can contain, such as excluding `null` elements. Adding an ineligible element will typically throw an unchecked `NullPointerException` or a `ClassCastException`.
```
Comment (Dusch4593) [inline content/java/concepts/set/set.md:14]:
```suggestion
A `Set` is an interface provided in the `java.util` package that implements collections with zero duplicate elements. Some implementations will have restrictions on the elements the `Set` can contain, such as excluding `null` elements. Adding an ineligible element will typically throw an unchecked `NullPointerException` or a `ClassCastException`.
```

File: content/java/concepts/set/set.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
+---
+Title: 'Set'
+Description: 'An interface for collections that contain no duplicate elements.'
```
Comment (Dusch4593) [inline content/java/concepts/set/set.md:3]:
```suggestion
Description: 'A Set is an interface for collections that contain zero duplicate elements.'
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Not too much from me for this one 😄 Review finished.

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented your suggestions.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM! 👍🏻

**Comment by KTom101** [comment]
@Dusch4593 Does this look good to merge with you? I have nothing else to add.

**Comment by Dusch4593** [comment]
PR merged! Live entry here: https://www.codecademy.com/resources/docs/java/set

**Pull request merged**